PromoExchange/batman
# Missing fields, correct shipping option on purchase screen and invoice email

See https://trello.com/c/L5L9ju9b

🏁 
- [x] Persist purchase model
- [x] Reference in purchase payment screen
- [x] Pass shipping option through to controller correctly
- [x] Send invoice email
- [x] Fix `Spree::Api::V1::ProductsController` reference

@spencerapplegate 
I think this is getting too big to carry on.  This PR is self contained as displaying missing fields and sending correct invoices.
The `best_price` plumbing is also included with PR but is transparent. If we get correct prices then that part is working. The outstanding issue is the credit card page will always display (and charge) the UPS price.  I will address this with a separate PR.
Total price is not exposed by the product.  I need to write a smaller wrapper around best price OR let `best_price` be a wrapper around `total_price`.
### Testplan
1. On any company store create a new sale
2. Ensure the product main color is displayed
3. Complete sale including credit card transaction
4. Ensure buyer and seller invoices are correct.
5. Check resque logs
6. Ensure resque starts correctly without a `Spree::Api::V1::ProductsController` reference error
